lib/puppet/module/task.rb

```diff
@@ -157,7 +185,9 @@ def implementations
     end
 
     def files
-      implementations.map {|imp| { 'name' => imp['name'], 'path' => imp['path'] } }
+      required_files = metadata.nil? ? [] : self.class.find_files(@name, metadata)
+      impls = implementations.map {|imp| { 'name' => imp['name'], 'path' => imp['path'] } }
+      impls + required_files
```

Comment:
This order is important (we should add a comment). pxp-agent is going to rely on impls being first.

Reply:
Commented